src/common/iop_order.c

```diff
@@ -1572,6 +1684,46 @@ int dt_ioppr_convert_onthefly(const int imgid)
   // the conversion happens.
   dt_image_write_sidecar_file(imgid);
 
+  return _iop_order_version;
+}
+
+// how is on-the-fly conversion done
+// Currently a hack to support v3/v4 history to later
+// returns the history version of imgid
+int dt_ioppr_convert_onthefly(const int imgid)
+{
+  int my_iop_order_version = 0;
+
+  // check current iop order version
+  sqlite3_stmt *stmt;
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "SELECT iop_order_version FROM main.images WHERE id = ?1",
+                              -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+  if(sqlite3_step(stmt) == SQLITE_ROW)
+  {
+    my_iop_order_version = sqlite3_column_int(stmt, 0);
+  }
+  sqlite3_finalize(stmt);
+
+  // already latest
+  if (my_iop_order_version == DT_IOP_ORDER_VERSION)
+  {
+    return my_iop_order_version;
+  }
+  else if (my_iop_order_version == 1)
+  {
+    // if an original image edited before iop-order was introduced we migrate it to v2
+    return _ioppr_migrate_iop_order(imgid, my_iop_order_version, 2);
+  }
+  else if (my_iop_order_version == 3 || my_iop_order_version == 4)
+  {
+    // ??? we handle only iop-version 3/4 (which have been broken) and move
+    // it to new v5.  this routine will be reused later when dt will
+    // propose in GUI a possibility to migrate old edits to a new
+    // version of iop-order.
+    return _ioppr_migrate_iop_order(imgid, my_iop_order_version, DT_IOP_ORDER_VERSION);
```

Comment:
Probably DT_IOP_ORDER_VERSION is right here instead of 5. Maybe leave a comment ?

Reply:
Done.